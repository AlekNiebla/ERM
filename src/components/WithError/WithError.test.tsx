import * as React from "react";
import { shallow } from "enzyme";
// components
import WithError from "./WithError";

describe("WithError component...", () => {
  it("...renders without errors", () => {
    const wrapper = shallow(<WithError />);

    expect(wrapper.state("hasError")).toBeFalsy();
    expect(wrapper.find(WithError)).toBeTruthy();
    // snapshot
    expect(wrapper).toMatchSnapshot();
  });
  it("...renders children", () => {
    const TestComponent = () => <span>Test</span>;
    const wrapper = shallow(
      <WithError>
        <TestComponent />
      </WithError>
    );

    expect(wrapper.find(TestComponent)).toBeTruthy();
  });
  describe("... handles errors by ...", () => {
    it("... calling componentDidCatch", () => {
      const TestComponent = () => <span>Test</span>;
      const wrapper = shallow(
        <WithError>
          <TestComponent />
        </WithError>
      );

      // mocking as jest function
      wrapper.instance().componentDidCatch = jest.fn();
      wrapper.simulateError(new Error());

      expect(wrapper.instance().componentDidCatch).toHaveBeenCalled();
    });
    it("... returning correct state by getDerivedStateFromError", () => {
      const result = WithError.getDerivedStateFromError();
      expect(result.hasError).toBe(true);
    });
  });
});
