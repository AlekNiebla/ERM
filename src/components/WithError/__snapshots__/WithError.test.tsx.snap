// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithError component... ... matches snapshot 1`] = `ShallowWrapper {}`;
